### DISCUSSION TITLE
HTTP request to grab files

### DISCUSSION THREAD

**Discussion by kentishh** [noahmaranesi/Ultimate-Guitar-Hack#1] [General]
Just came across your post on reddit which I found interesting. Turns out, it's really easy to grab the files you are after from a terminal using cURL. Here is an example, downloading the Killing In the Name tab:

curl --referer https://tabs.ultimate-guitar.com/tab/rage-against-the-machine/killing-in-the-name-official-2238297  https://tabs.ultimate-guitar.com/tab/download?id=2238297 -o filenamehere.gpx

As long as you include the referer header in your GET request, you can pull down the file. 

I found it quite difficult to actually get to the page with the tab without being bombarded with sign up requests etc. How did you manage to find the ID of this tab?

Cheers!

**Comment by noahmaranesi** [comment]
Hey @kentishh! Thanks for that! I knew there was definitely some easier way to go about it.

The easiest way to find an ID for a tab is via its URL! (See Step 7)

**Reply by kentishh** [reply]
No problem! 

Ah I see. For example, if I try to open one of the "official" tabs, I just get redirected to a signup page to sign up to the pro version. Do you know if there is a way to bypass that at all? I get redirected here when trying to open one of the official tabs: 
```
https://www.ultimate-guitar.com/pro/?utm_source=UltimateGuitar&utm_medium=Search&utm_campaign=UG+Search&utm_content=Official+Version&artist=Rage%20Against%20the%20Machine&song=killing%20in%20the%20name
```

If there is no way to bypass this, then I imagine you'd never be able to actually get the unique ID from the URL.

**Reply by noahmaranesi** [reply]
That's true. I didn't actually test *getting* a URL (since I already had it). Unfortunately I am unable to retrieve the tab ID (URL in general, as you mentioned) through Ultimate Guitar's native search. Sorry for any confusion!
